### PR TITLE
Convert ignition_simple to C++

### PR DESCRIPTION
**Pull request by maxpkatz**

### CODE REVIEW

**Comment by shardi2** [comment]
Mass_Fractions_plots_ignition

**Comment by shardi2** [comment]
Mass_Fractions_plots_ignition

**Comment by shardi2** [comment]
Mass_Fractions_plots_ignition

File: networks/ignition_simple/actual_rhs.H [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    Real dT9adt = (T9a / T9 - (T9a / (1.0_rt + 0.0396_rt * T9)) * 0.0396_rt) * dT9dt;
+
+    // compute the CF88 rate
+    Real scratch    = std::pow(T9a, (1.0_rt / 3.0_rt));
+    Real dscratchdt = (1.0_rt / 3.0_rt) * std::pow(T9a, (-2.0_rt / 3.0_rt)) * dT9adt;
+
+    Real a       = 4.27e26_rt * std::pow(T9a, (5.0_rt / 6.0_rt)) * std::pow(T9, -1.5e0_rt);
+    Real dadt    = (5.0_rt / 6.0_rt) * (a / T9a) * dT9adt - 1.5e0_rt * (a / T9) * dT9dt;
+
+    Real b       = exp(-84.165e0_rt / scratch - 2.12e-3_rt * T9 * T9 * T9);
```
Comment (zingale) [inline networks/ignition_simple/actual_rhs.H:52]: std::exp

File: networks/ignition_simple/actual_rhs.H [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    // Now get the data from the state.
+
+    Real temp = state.T;
+    Real dens = state.rho;
+
+    Array1D<Real, 1, NumSpec> y;
+    for (int i = 1; i <= NumSpec; ++i) {
+        y(i) = state.xn[i-1] * aion_inv[i-1];
+    }
```
Comment (zingale) [inline networks/ignition_simple/actual_rhs.H:107]: y is never used

**Comment by zingale** [comment]
test_rhs output:
```

            variable name            absolute error            relative error
                                        (||A - B||)         (||A - B||/||A||)
 ----------------------------------------------------------------------------
 level = 0
 density                                          0                         0
 temperature                                      0                         0
 Ydot_carbon-12                     2.384185791e-07           1.536837808e-16
 Ydot_oxygen-16                                   0                         0
 Ydot_magnesium-24                  1.192092896e-07           1.536837808e-16
 Xold_carbon-12                                   0                         0
 Xold_oxygen-16                                   0                         0
 Xold_magnesium-24                                0                         0
 Tdot                                        745472           1.121330205e-14
 Edot                               2.199023256e+12           2.108979404e-16
 J_carbon-12_carbon-12               1.14440918e-05           2.766308055e-16
 J_oxygen-16_carbon-12                            0                         0
 J_magnesium-24_carbon-12            1.14440918e-05           2.766308055e-16
 J_T_carbon-12                             19922944           1.123794667e-14
 J_E_carbon-12                      7.036874418e+13           2.530775285e-16
 J_carbon-12_oxygen-16                            0                         0
 J_oxygen-16_oxygen-16                            0                         0
 J_magnesium-24_oxygen-16                         0                         0
 J_T_oxygen-16                                    0                         0
 J_E_oxygen-16                                    0                         0
 J_carbon-12_magnesium-24                         0                         0
 J_oxygen-16_magnesium-24                         0                         0
 J_magnesium-24_magnesium-24                         0                         0
 J_T_magnesium-24                                 0                         0
 J_E_magnesium-24                                 0                         0
 J_carbon-12_T                       1.33226763e-15           3.455080587e-16
 J_oxygen-16_T                                    0                         0
 J_magnesium-24_T                                 0                         0
 J_T_T                               0.001800537109           1.089639972e-14
 J_E_T                                         5120            1.97556463e-16
 J_carbon-12_E                                    0                         0
 J_oxygen-16_E                                    0                         0
 J_magnesium-24_E                                 0                         0
 J_T_E                                            0                         0
 J_E_E                                            0                         0
```

File: networks/ignition_simple/actual_rhs.H (hunk 1 above)
Comment (maxpkatz) [reply networks/ignition_simple/actual_rhs.H]: Fixed

File: networks/ignition_simple/actual_rhs.H (hunk 2 above)
Comment (maxpkatz) [reply networks/ignition_simple/actual_rhs.H]: Fixed

**Comment by zingale** [comment]
this looks good to me.  Here's my burn_cell test (I set X(C12) = 0.75, X(O16) = 0.25
ignition_c12

**Pull request merged**